### PR TITLE
[WIP] Correct inspect options

### PR DESCRIPTION
**Pull request by eksperimental**
This PR partially solves https://github.com/elixir-lang/elixir/pull/4582#discussion_r62423420

@josevalim: I haven't implemented your suggestion for checking valid values for the options.
I think we should come up with a mechanism something that can help us check all values, otherwise it's gonna be a lot of code duplication.

regarding keys, a simple struct!/1 will do it (i think), let me know if this works for you.

### CODE REVIEW

File: lib/elixir/lib/kernel.ex [hunk 1]
```diff
@@ -1570,7 +1570,7 @@ defmodule Kernel do
   """
   @spec inspect(Inspect.t, Keyword.t) :: String.t
   def inspect(arg, opts \\ []) when is_list(opts) do
-    opts  = struct(Inspect.Opts, opts)
+    opts  = struct!(Inspect.Opts, opts)
```
Comment (josevalim) [inline lib/elixir/lib/kernel.ex]: backwards incompatible change. inspect will now raise if the options do not always match, we should probably revert this. :)

Comment (eksperimental) [inline lib/elixir/lib/kernel.ex]: so, should we warn? and use try/rescue?

Comment (josevalim) [inline lib/elixir/lib/kernel.ex]: No, i think it is fine. Most of the Elixir APIs won't warn if you pass extra options so it does not force you to always prune it.

Comment (eksperimental) [inline lib/elixir/lib/kernel.ex]: fixed

Comment (WizardOfOgz) [inline lib/elixir/lib/kernel.ex]: @josevalim Should there be a test to prevent this regression?

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**